protocols/xml/src/main/java/org/opennms/protocols/xml/collector/Sftp3gppXmlCollectionHandler.java

```diff
@@ -76,6 +78,20 @@ public XmlCollectionSet collect(CollectionAgent agent, XmlDataCollection collect
         collectionSet.setCollectionTimestamp(new Date());
         collectionSet.setStatus(ServiceCollector.COLLECTION_UNKNOWN);
 
+        // Create a map of parameters from the service to use in URLs, etc.
+        Map<String, String> params = new HashMap<>();
+        for(Map.Entry<String, Object> entry : parameters.entrySet()) {
+            if(entry.getKey().equals("SERVICE"))
+                continue;
+            if(entry.getKey().equals("collection"))
+                continue;
+            if(entry.getKey().equals("xml-collection"))
+                continue;
+            if(entry.getKey().equals("handler-class"))
+                continue;
+            params.put(entry.getKey(), ParameterMap.getKeyedString(parameters, entry.getKey(), ""));
```

Comment:
This seems to be duplicated code, as it is used multiple times. Can we create a filter function or something like that? e.g:

``` java
// TODO add javadoc
// TODO verify that it actually works
public Map<String, String> filterParameters(Map<String, Object> input) {
   Set<String> blackList = Sets.newHashSet("SERVICE", "collection", "xml-collection", "handler-class");
   return   input.entrySet().stream()
                      .filter(e -> !blackList.contains(e.getKey())
                      .collect(Collectors.toMap(e -> e.getKey(), e -> ParameterMep.getKeyedString(input, e.getKey(), ""));
}
```